### ISSUE TITLE
Table renders in "mobile" format even on desktop

### ISSUE THREAD

**Issue by pdpinch** [mitodl/ocw-hugo-themes#835]
### Steps to Reproduce

Visit https://draft.ocw.mit.edu/courses/res-3-006-micro-mentoring-resources-and-materials-science-curriculum-spring-2021/pages/application-evaluation-rubric/ to see the table that will not format correctly. Studio: https://ocw-studio.odl.mit.edu/sites/micro-mentoring-resources-and-materials-science-curriculum/type/page/edit/80e34d94-aa1f-45ba-ad92-625b174ae5db/

### Expected Behavior

Tables should be formatted for desktop display, with rows and columns.

### Actual Behavior

Table is rendered in mobile "stacked" format even at desktop widths. 

### Screenshot or Screencast

![Screen Shot 2022-08-30 at 9 20 23 AM](https://github.mit.edu/storage/user/3602/files/4f800d80-2845-11ed-8847-1d649df97449)

**Comment by pdpinch** [comment]
What does this page look like in the course-v2 theme?

**Comment by Wassaf-Shahzad** [comment]
@pdpinch 

### Tables in Course-V2
<img width="1440" alt="Screenshot 2022-09-01 at 1 19 49 PM" src="https://user-images.githubusercontent.com/87968618/187869814-c9db0bd2-6350-4ee6-8d7b-ae888ab5d280.png">

The reason the table is rendered in a mobile view because of [this line here](https://github.com/mitodl/ocw-hugo-themes/blob/main/course/assets/js/responsive_tables.js#L25)
As you can see in the above image if the table width is greater than the content width then the table is not rendered correctly

**Comment by pdpinch** [comment]
> The reason the table is rendered in a mobile view because of [this line here](https://github.com/mitodl/ocw-hugo-themes/blob/main/course/assets/js/responsive_tables.js#L25) 

Ok, that makes sense. So the problem is the content has 5 (!) columns? One thing I can do is talk to the author about reconsidering the layout. 

The author is also complaining that the rendering is inconsistent. Is that because this code is client-dependent? Is there a way we can make that behavior more predictable? 

I see you opened a PR, but I don't understand what it does.

**Comment by Wassaf-Shahzad** [comment]
> > The reason the table is rendered in a mobile view because of [this line here](https://github.com/mitodl/ocw-hugo-themes/blob/main/course/assets/js/responsive_tables.js#L25)
> 
> Ok, that makes sense. So the problem is the content has 5 (!) columns? One thing I can do is talk to the author about reconsidering the layout.
> 
> The author is also complaining that the rendering is inconsistent. Is that because this code is client-dependent? Is there a way we can make that behavior more predictable?
> 
> I see you opened a PR, but I don't understand what it does.

### The reason for inconsistency 

The reason for inconsistency is the that code actually checks the width for the whole body rather than the `main content`. which can lead to some inconsistency.

### What the PR fixes

The PR fixes the above mention inconsistency but only in course-v2. Rather than checking with the body it checks the width of the main tag.

**Comment by gumaerc** [comment]
@Wassaf-Shahzad @pdpinch The code currently checks the width of the div with the ID `main-content`:

```
    // force mobile style if table width exceeds main content area
    const tableWidth = table.clientWidth
    const mainContentWidth = document.getElementById("main-content").clientWidth
    if (tableWidth > mainContentWidth) {
      table.classList.add("mobile-table")
    }
```

The `main-content` div refers to the page content area:
![image](https://user-images.githubusercontent.com/12089658/188929144-9ada6515-1c2c-49d8-92ff-08261d7619e9.png)

I think the main issue here is that this code is only run at page load, and if the page loads with the course info drawer open that width is much narrower than the actual available space if the drawer is closed. I think we should:

 - Re-run the check and apply / remove the mobile table style when opening / closing the course info drawer.
 - If a table exists on the page and the content area is not wide enough to display it, default to having the course info drawer closed. Ideally we should only do this if it will be enough to display the table with desktop styles, but I'm not sure this kind of check will be possible without layout thrashing.

**Comment by pdpinch** [comment]
> Re-run the check and apply / remove the mobile table style when opening / closing the course info drawer.

Yes, there's an issue for this #401. But I don't see how this would help fix this issue?

> If a table exists on the page and the content area is not wide enough to display it, default to having the course info drawer closed. Ideally we should only do this if it will be enough to display the table with desktop styles, but I'm not sure this kind of check will be possible without layout thrashing.

If there's a way to do this, I would be OK with it.

**Comment by Wassaf-Shahzad** [comment]
@gumaerc @pdpinch, I checked on course-v2 and it seems that `course-main-content` covers neatly the whole page and `course-content-section` covers the content section.
Also should I merge the PR ? 

![Screenshot 2022-09-08 at 5 55 06 PM](https://user-images.githubusercontent.com/87968618/189127163-949a5488-878c-4969-bd33-e2ea80bc4447.png)

**Comment by gumaerc** [comment]
I see. Sorry I wasn't looking at `course-v2`, my mistake. I'll leave a comment on the PR.

**Comment by pdpinch** [comment]
Can this be closed after https://github.com/mitodl/ocw-hugo-themes/pull/844 merges?

**Comment by Wassaf-Shahzad** [comment]
> Can this be closed after #844 merges?

Yes I believe
